src/Facets/HopFacetOptimized.sol

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { IHopBridge } from "../Interfaces/IHopBridge.sol";
+import { LibAsset, IERC20 } from "../Libraries/LibAsset.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+import { SwapperV2, LibSwap } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+
+/// @title Hop Facet (Optimized)
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through Hop
+contract HopFacetOptimized is ILiFi, SwapperV2, Validatable {
+    /// Types ///
+
+    struct HopData {
+        uint256 bonderFee;
+        uint256 amountOutMin;
+        uint256 deadline;
+        uint256 destinationAmountOutMin;
+        uint256 destinationDeadline;
+        IHopBridge hopBridge;
+    }
+
+    /// Events ///
+
+    event HopBridgeRegistered(address indexed assetId, address bridge);
+
+    /// External Methods ///
+
+    /// @notice Sets approval for the Hop Bridge to spend the specified token
+    /// @param bridges The Hop Bridges to approve
+    /// @param tokensToApprove The tokens to approve to approve to the Hop Bridges
+    function setApprovalForBridges(address[] calldata bridges, address[] calldata tokensToApprove) external {
+        for (uint256 i; i < bridges.length; i++) {
+            // Give Hop approval to bridge tokens
+            LibAsset.maxApproveERC20(IERC20(tokensToApprove[i]), address(bridges[i]), type(uint256).max);
+        }
+    }
+
+    /// @notice Bridges ERC20 tokens via Hop Protocol from L1
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _hopData data specific to Hop Protocol
+    function startBridgeTokensViaHopL1ERC20(ILiFi.BridgeData calldata _bridgeData, HopData calldata _hopData)
+        external
+        validateBridgeData(_bridgeData)
+    {
+        // Deposit assets
+        LibAsset.depositAsset(_bridgeData.sendingAssetId, _bridgeData.minAmount);
```

Comment:
The `depositAsset` function checks if the user actually has the balance we want to transfer, instead I would call a transfer right away and it will simply fail if the user does not has the balance or has not approved the token yet.

Reply:
Fixed

Reply:
How did you fix it? I dont see any changes in LibAsset and you are still calling the library (rather than doing the transfer straight away here? Is this topic really fixed?